src/dawn_native/d3d12/DeviceD3D12.cpp

```diff
@@ -169,12 +169,14 @@ namespace dawn_native { namespace d3d12 {
         mD3d12Device = nullptr;
         mCommandQueue = nullptr;
 
-        delete mCommandAllocatorManager;
-        delete mDescriptorHeapAllocator;
-        delete mMapRequestTracker;
-        delete mResourceAllocator;
-        delete mResourceUploader;
-        delete mFunctions;
+        mCommandAllocatorManager = nullptr;
+        mDescriptorHeapAllocator = nullptr;
+        mMapRequestTracker = nullptr;
+        mResourceAllocator = nullptr;
+        mResourceUploader = nullptr;
+
+        // Delete mFunctions last since it unloads the D3D12 DLLs
```

Comment:
I believe we could move this comment to the field order, and remove all the explicit deletes here. https://stackoverflow.com/a/2254306

Reply:
TIL! Done, this makes the code much prettier :D